### DISCUSSION TITLE
Difference between ZPL_STATIC and ZPL_STATIC_BUILD?

### DISCUSSION THREAD

**Discussion by ib00** [zpl-c/zpl#96] [Q&A] [answered]
I see an option 'ZPL_STATIC' in zpl.h. When is this actually used?

**Comment by zpl-zak** [comment, marked as answer]
ZPL_STATIC_LIB is used when you intend to compile zpl into its own .lib file.

Whereas ZPL_STATIC enforces static linkage of symbols within compilation units that define ZPL_IMPLEMENTATION, thus ensuring symbols don't leak outside the unit. This is useful if you intend to compile zpl with different compile-time switches in multiple source files, each relying on ZPL_IMPLEMENTATION to pull the code in and avoid having symbol collisions in between them.

Sorry for the very late response!
